### PR TITLE
e2e: refactor TLS helpers

### PR DESCRIPTION
**Pull request by stlaz**
- simplification of the current certgen code
- better separation of leaf and CA certs in kube CMs/Secrets

### CODE REVIEW

File: test/kubetest/kubernetes.go [hunk 1]
```diff
@@ -79,12 +73,16 @@ func createCerts(client kubernetes.Interface, name string, createSignedCert cert
...
 		})
 
-		secretName := fmt.Sprintf("%s-keys", name)
+		secretName := fmt.Sprintf("%s-certs", name)
 		_, err = client.CoreV1().Secrets(ctx.Namespace).Create(context.TODO(), &corev1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: secretName,
 			},
 			Data: map[string][]byte{
+				"tls.crt": pem.EncodeToMemory(&pem.Block{
```
Comment (alegacy) [inline test/kubetest/kubernetes.go:82]: 👍

**Review by alegacy — APPROVED** [review]
/lgtm

...thought maybe there was an issue since the tests failed, but ran them on my system and they were fine.

**Comment by stlaz** [comment]
The test failed on KRP logging
`I1126 13:37:03.550424 1 log.go:245] http: proxy error: dial tcp 10.96.218.31:8443: i/o timeout`
That might be transient. The tests seemed to pass on my machine, too :eyes:

**Comment by stlaz** [comment]
Still not able to reproduce locally. Raising nginx loglevel to debug in CI.

**Comment by stlaz** [comment]
Looks like the test was missing assumptions on when it's safe to be run. I'll give it a couple respins to check the timing issue is now gone.

**Pull request merged**